packages/zendesk-adapter/src/filters/article/article.ts

```diff
@@ -234,11 +235,69 @@ const handleArticleAttachmentsPreDeploy = async ({ changes, client, elementsSour
 const getInline = (attachment: InstanceElement | undefined): boolean => attachment?.value.inline
+const shouldRemoveAttachment = (attachment: Element, allRemovedAttachmentsIds: Set<number>): boolean =>
+  isInstanceElement(attachment) && allRemovedAttachmentsIds.has(attachment.value.id)
 
+const isRemovedAttachment = (
+  attachment: unknown,
+  articleRemovedAttachmentsIds: Set<number>,
+  attachmentByName: Record<string, InstanceElement>
+): boolean => {
+  const name = isReferenceExpression(attachment) ? getName(attachment) : undefined
+  const attachmentInstance = name ? attachmentByName[name] : undefined
+  return isInstanceElement(attachmentInstance)
+    && articleRemovedAttachmentsIds.has(attachmentInstance.value.id)
+}
+
+const getAttachmentData = (
+  article: InstanceElement,
+  attachmentByName: Record<string, InstanceElement>
+): {inlineAttachmentsNameById: Record<number, string>; attachmentIdsFromArticleBody: Set<number | undefined>} => {
+  const attachmentsNames: string[] = article.value.attachments?.filter(isReferenceExpression).map(getName)
+  const inlineAttachmentInstances = attachmentsNames
+    .map(name => attachmentByName[name])
+    .filter(isInstanceElement)
+    .filter(attachment => attachment.value.inline)
+  const inlineAttachmentsNameById = _.mapValues(_.keyBy(inlineAttachmentInstances, getId), getName)
+  const articleBody = article.value.body ?? ''
+  const attachmentIdsFromArticleBody = new Set(
+    Array.from(articleBody.matchAll(ATTACHMENTS_IDS_REGEX), (match: RegExpMatchArray | undefined) => match?.groups?.id)
+      .filter(id => id !== undefined).map(Number)
+  )
+  return { inlineAttachmentsNameById, attachmentIdsFromArticleBody }
+}
+
+/**
+ * returns deleted attachments ids
+ */
+const calculateAndRemoveDeletedAttachments = (
+  articleInstances: InstanceElement[],
+  attachmentByName: Record<string, InstanceElement>
+): Set<number> => {
+  const allRemovedAttachmentsIds = new Set<number>()
+  const allRemovedAttachmentsNames = new Set<string>()
+  articleInstances.forEach(article => {
+    const articleRemovedAttachmentsIds = new Set<number>()
+    const attachmentData = getAttachmentData(article, attachmentByName)
+    Object.keys(attachmentData.inlineAttachmentsNameById).forEach(id => {
+      const numberId = Number(id)
+      if (!attachmentData.attachmentIdsFromArticleBody.has(numberId)) {
+        allRemovedAttachmentsIds.add(numberId)
+        articleRemovedAttachmentsIds.add(numberId)
+        allRemovedAttachmentsNames.add(attachmentData.inlineAttachmentsNameById[numberId])
+      }
+    })
+    article.value.attachments = article.value.attachments?.filter(
+      (attachment: unknown) => !isRemovedAttachment(attachment, articleRemovedAttachmentsIds, attachmentByName)
+    )
+  })
+  log.info(`the article attachments ${Array.from(allRemovedAttachmentsNames)} are going to be deleted since they are inline but do not appear in the body`)
```

Comment:
nit: this list can be long, so might be better to finish the sentence first - 
```suggestion
  log.info(`the following article attachments are going to be deleted, since they are inline but do not appear in the body: ${Array.from(allRemovedAttachmentsNames)} `)
```
(also - maybe "not going to be included in the fetch" rather than "deleted", just to avoid unnecessary alarm for in the future? :) )